src/library/Flex/Flex.js

```diff
@@ -159,33 +221,31 @@ const Root = createStyledComponent(Box, styles.root, {
 });
 
 const ThemedRoot = withTheme(
-  ({ breakpoints, children, gutterWidth, theme, ...restProps }) => {
+  ({ breakpoints, children, direction, gutterWidth, theme, ...restProps }) => {
     const rootProps = {
       breakpoints,
+      direction,
       gutterWidth,
       ...restProps
     };
 
     let flexItems;
-    flexItems = Children.map(children, (child) => {
-      const {
-        breakpoints: propBreakpoints,
-        margin: propMargin,
-        marginHorizontal,
-        marginLeft,
-        marginRight
-      } = child.props;
+    flexItems = Children.map(children, (child, index) => {
+      const { breakpoints: propBreakpoints, ...restChildProps } = child.props;
       let props = {
         breakpoints: propBreakpoints || breakpoints
       };
 
-      if (gutterWidth) {
-        const margin =
-          marginHorizontal || propMargin || getGutterSize(theme, gutterWidth);
+      const flexItemsCount = Children.count(children);
```

Comment:
Can this be moved out of the iteration so it only runs once?

Reply:
Good call. I had left it inside the `if (child.props)` conditional to avoid the work, but it's better to do it a possible extra one time rather than `n - 1` times.